src/poetry/utils/env.py

```diff
@@ -456,6 +457,30 @@ class EnvManager:
     def __init__(self, poetry: "Poetry") -> None:
         self._poetry = poetry
 
+    def _detect_active_python(self, io: "IO") -> str:
+        executable = None
+
+        try:
+            io.write_line(
+                "Trying to detect current active python executable as specified in the config.",
+                verbosity=Verbosity.VERBOSE,
+            )
+            executable = decode(
+                subprocess.check_output(
+                    list_to_shell_command(
+                        ["python", "-c", '"import sys; print(sys.executable)"']
```

Comment:
This is not necessarily universal.

Reply:
You mean `python`? In theory yes. 

But if you have e.g. `python2` and `python3` in your shell but no `python` it is impossible to decide what python the user expects.

The main use case for the implementation here is for users who switch there `python` to different versions (e.g. with pyenv). I'm not aware of other tools, but I guess they all will provide a `python`. As a fallback if no `python` is found and the option is activated, the default behavior will be used.

The only generic solution for all edge cases I can imagine right now, is to let the user set a "default python executable" that should be used. But in this case I have some security concerns. Running a user provided command listed in a config, doesn't sound good.

Reply:
This is not just about `python2` vs `python3`. There also are other issues like for `.exe` extensions on windows etc depending on where `poetry` is running. Additionally, quite a lot of tools rely on environment magic to wrangle which `python` version to use etc., this can also be impacted by how the subprocess shell is spawned (eg: is it a login shell?, how does the tool configure the "current" python?, are profile scripts required? is the `PATH` discoery set correctly? etc.). There are too many variables here. Not sure if this is a show stopper, but we should really consider if this makes things more obscure.

An additional point might be that we should fall back gracefully to compatible version detection than crashing as the option is "prefer" and not "use".

Reply:
> An additional point might be that we should fall back gracefully to compatible version detection than crashing as the option is "prefer" and not "use".

This is already the case :)